### PR TITLE
feat: Receiving Mentions

### PR DESCRIPTION
**Pull request by alexandreferris**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [X] contains a reference JIRA issue number like `SQPIT-764`
    - [X] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

[AR-2545 - Receiving @mentions](https://wearezeta.atlassian.net/browse/AR-2545)

### Issues

Mentions were not being properly highlighted in a message.

### Causes (Optional)

Mentions were not being handled and mapped down to UI layers.

### Solutions

Map mentions down until UI layer and display them accordingly.

#### How to Test

- Open App
- Open a conversation
- Receive messages with mentions (yourself or another user)
- Click on the mention:
   - Yourself => Your profile screen
   - Other User => Other User profile screen 🙃 

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 
| Before | After |
| ----------- | ------------ |
| bad_mentions | good_mentions |

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1125/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1125/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4798517)(4798517)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4798517)(4798517)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/theme/WireColorPalette.kt [hunk 1]
```diff
@@ -152,4 +152,7 @@ object WireColorPalette {
     @Stable val Gray100 = Color(0xFF17181A)
 
     @Stable val GrayRed900  = Color(0xFF390000)
+
+    @Stable val messageMentionBackground = Color(0x1A0772DE)
+    @Stable val messageMentionText = Color(0xFF0772DE)
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/theme/WireColorPalette.kt:157]: Not really sure if this should go here, but couldn't find this color in our code and didn't really find anywhere else to add them as well

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1125/4/) **failed**.

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1125/5/) **failed**.

**Review by gongracr — APPROVED** [review]
Great job. Tried on my emulator and worked like a charm 👌🏼🍝

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-1125/7/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4824104)(4824104)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4824104)(4824104)-dev-release.apk)

**Pull request merged**